### ISSUE TITLE
[HELP] How to print MMU related information on ARM FVP?

### ISSUE THREAD

**Issue by lzc1001** [apache/nuttx#13062]
### Description

I want to print virtual address and physical address when initialize MMU in arm64_mmu.c. So, I set CONFIG_MMU_DEBUG = 1 and set other configuration as following picture.
![image](https://github.com/user-attachments/assets/6c897f16-ce7f-494e-ad74-f3af7aaa431a)
But the related mmu configuration information do not print. 
Thanks for everyone!

### Verification

- [X] I have verified before submitting the report.

**Comment by acassis** [comment]
@lzc1001 look inside arch/arm64/src/common/arm64_mmu.c the virtual and physical addresses are no printed by default, you need to add a new "sinfo()" with that values you want to print

**Comment by lzc1001** [comment]
@acassis Thanks for your answer!😀 So, I need to write a new "sinfo()" by myself? Because I found that this function was already written in the code, but it was wrapped in the macro definition.
![image](https://github.com/user-attachments/assets/c5d3261e-c0d1-4f3a-a30f-53bdfbec7398)

**Comment by acassis** [comment]
@lzc1001 hmm, I think I discovered the issue: there is not CONFIG_MMU_DEBUG.
Just run: $ git grep MMU_DEBUG

There need to exist an MMU_DEBUG symbol defined inside a Kconfig or you can just comment out these lines:

```
/* MMU debug option
 * #define CONFIG_MMU_ASSERT 1
 * #define CONFIG_MMU_DEBUG 1
 * #define CONFIG_MMU_DUMP_PTE 1
 */
```

**Comment by acassis** [comment]
@lzc1001 if you get it working, please close this Issue

**Comment by lzc1001** [comment]
@acassis i add "#define CONFIG_MMU_DEBUG 1" to arch/arm64/src/common/arm64_mmu.c, then run on FVP. But the related mmu information still do not print.😫

**Comment by acassis** [comment]
@lupyuen any idea?

**Comment by lupyuen** [comment]
Hi @lzc1001 you mentioned FVP, could you verify that this is the Build Config that we used?

```bash
$ tools/configure.sh fvp-armv8r:nsh
```

`fvp-armv8r:nsh` doesn't seem to support MMU. Maybe that's why the MMU Logging wasn't working?

```bash
$ grep MMU .config
# CONFIG_ARCH_HAVE_MMU is not set
```

I tried `qemu-armv8a:nsh`, but it doesn't support MMU either:

```bash
$ ./tools/configure.sh qemu-armv8a:nsh
$ grep MMU .config
CONFIG_ARCH_HAVE_MMU=y
# CONFIG_ARCH_USE_MMU is not set
```

**Comment by lzc1001** [comment]
@lupyuen Thanks for your answer! I ported NuttX to A53 based on FVP in the last few days and ARM A53 have MMU. Then, NuttX can boot and input command to terminal, like "ls", "hello", etc. So, I think MMU is enabled for NuttX on FVP. But I do't make sure that the porting code have some potential bug, hhh😀.
I add code to arch/arm64/src/common/arm64_mmu.c and enable the related congfiguration by menuconfig. But the MMU infnormation still do not print. Maybe porting code have other issues. I will check my added code. The NuttX commit ID is 7a865b247969a9dbc06eddcaf58c59a14dd05d6d, tag: nuttx-12.6.0-RC1
![image](https://github.com/user-attachments/assets/611425bd-3e89-4e0e-ba06-99147a830c60)
![image](https://github.com/user-attachments/assets/d54fc950-f283-499d-9d93-ddd322dfe6b6)

**Comment by lupyuen** [comment]
@lzc1001 Can you verify that the NuttX Config supports MMU?

```bash
$ grep MMU .config
CONFIG_ARCH_HAVE_MMU=y
# CONFIG_ARCH_USE_MMU is not set
```

Is it possible that the MMU code is disabled due to other reasons? Perhaps we can add an Error Message in arch/arm64/src/common/arm64_mmu.c:

```c
int arm64_mmu_init(bool is_primary_core)
{
#error MMU is enabled
...
```

**Comment by lzc1001** [comment]
@lupyuen Yes! The config support MMU
![image](https://github.com/user-attachments/assets/f98e4ab9-9df1-4f01-ac8d-7ce8f277acfb)

**Comment by lupyuen** [comment]
@lzc1001 I wonder if `sinfo` is failing to log correctly. What if you rename all `sinfo` to `_info`?

**Comment by raiden00pl** [comment]
`arm64_mmu_init()` is called early in `arm64_chip_boot()` so it's possible that serial console is not yet initialized. `arm64_earlyserialinit()` is called after `arm64_mmu_init()`. You can add simple debug message (`_info("xxxx")`) at the top of `arm64_chip_boot()` and after `arm64_mmu_init()` to see if serial port works at this boot stage.

**Comment by acassis** [comment]
@raiden00pl @lupyuen @lzc1001 maybe RAMLOG could solve this issue.

**Comment by raiden00pl** [comment]
@acassis RAMLOG is registered also late in boot process (in `drivers_initialize()`). One option if serial is not available since boot and requires some kind of initialization is to separate `arm64_mmu_init()` from MMU debug dump functionality like it's done for ACPI in x86:
1. ACPI is parsed in early boot stage in `__nxstart()` as it's required for other initialization tasks
2. ACPI dump is done in `up_initialize()` when some low-level initialization is done (like `x86_64_serialinit()`)

**Comment by lzc1001** [comment]
@acassis @raiden00pl @lupyuen Thank you very much! 
I added simple debug message (_info("111111111111111\n")). Then, the MMU information still don't print. but I added "_info("1111111111111111\n");" before and after up_initialize() in nuttx/sched/init/nx_start.c, the debug information after up_initialize() do print!! hhh, you are very niubility!😀✌
I tried to use ARM TF-A to boot NuttX and added the related macro in arch/arm64/src/common/arm64_mmu.c, the terminal print the related MMU information. But the exact cause of this problem is still unknown, I will continue to check my poring code!
Thank you very much again!
